oauth2_provider/tests/test_oauth2_backends.py

```diff
@@ -1,7 +1,56 @@
-from django.test import TestCase, RequestFactory
+import json
 
+from django.test import TestCase, RequestFactory
 
 from ..backends import get_oauthlib_core
+from ..oauth2_backends import OAuthLibCore, JSONOAuthLibCore
+
+
+class TestOAuthLibCoreBackend(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.oauthlib_core = OAuthLibCore()
+
+    def test_form_urlencoded_extract_params(self):
+        payload = "grant_type=password&username=john&password=123456"
+        request = self.factory.post("/o/token/", payload, content_type="application/x-www-form-urlencoded")
+
+        uri, http_method, body, headers = self.oauthlib_core._extract_params(request)
+        self.assertIn("grant_type=password", body)
+        self.assertIn("username=john", body)
+        self.assertIn("password=123456", body)
+
+    def test_application_json_extract_params(self):
```

Comment:
This test enforces that the default backend class doesn't accept an ``application/json`` Content-Type